command/agent/template/template_test.go

```diff
@@ -127,13 +160,27 @@ func TestServerRun(t *testing.T) {
 			if ts == nil {
 				t.Fatal("nil server returned")
 			}
+			server.testingLimitRetry = true
 
-			go server.Run(ctx, templateTokenCh, templatesToRender)
+			errCh := make(chan error, 1)
+			go server.Run(ctx, templateTokenCh, templatesToRender, errCh)
 
 			// send a dummy value to trigger the internal Runner to query for secret
 			// info
 			templateTokenCh <- "test"
-			<-server.DoneCh
+
+			select {
+			case <-server.DoneCh:
```

Comment:
Can `DoneCh` be closed/have a value at the same time the error channel has one? Could this cause flakiness?

Reply:
I don't think so since the error is sent back before the DoneCh is closed on a defer.

Reply:
I think @pcman312 is right, if there are values on both `DoneCh` and `errCh` then go will randomly choose one of them. You can't be sure that select will choose the expected thing here.